ext/cache/src/main.rs

```diff
@@ -0,0 +1,232 @@
+// main.rs
+
+// Copyright (C) 2020-2021 The Nitrocli Developers
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+mod ext;
+
+use std::collections;
+use std::fs;
+use std::io;
+use std::path;
+
+use anyhow::Context as _;
+
+type OtpCache = collections::BTreeMap<String, Vec<Slot>>;
+
+#[derive(Debug, Default, serde::Deserialize, serde::Serialize)]
+struct PwsCache {
+  slots: Vec<Slot>,
+}
+
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+struct Slot {
+  name: String,
+  id: usize,
+}
+
+/// Access Nitrokey OTP slots by name
+#[derive(Debug, structopt::StructOpt)]
+#[structopt(bin_name = "nitrocli cache")]
+struct Args {
+  #[structopt(subcommand)]
+  cmd: Command,
+}
+
+#[derive(Debug, structopt::StructOpt)]
+enum Command {
+  /// Access the OTP cache
+  Otp(OtpCommand),
+  /// Access the PWS cache
+  Pws(PwsCommand),
```

Comment:
Do you think you could split the commit in two: one for the OTP functionality and another for PWS stuff? (if for nothing else than for reviewing changes and just having tangentially smaller sized commits)